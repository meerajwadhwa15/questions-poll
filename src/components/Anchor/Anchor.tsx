import React, { FunctionComponent } from "react";
import { Link } from "react-router-dom";
export type Props = {
  to: string;
  title: string;
  children: React.ReactNode;
  className?: string;
  onClick?: () => void;
  rel?: string;
};

const Anchor: FunctionComponent<Props> = ({ children, ...others }) => (
  <Link {...others}>{children}</Link>
);

export default Anchor;
