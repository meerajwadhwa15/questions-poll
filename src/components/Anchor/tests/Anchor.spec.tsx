import React from "react";
import Anchor from "../Anchor";
import { create } from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

const props = {
    to: '/questions/1',
    title: 'Question1'
};

test("Matches the snapshot with Anchor", () => {
  const component = create(
    <BrowserRouter>
      <Anchor {...props}>Question1</Anchor>
    </BrowserRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
