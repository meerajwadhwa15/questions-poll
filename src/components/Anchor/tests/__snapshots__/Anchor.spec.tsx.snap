// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot with Anchor 1`] = `
<a
  href="/questions/1"
  onClick={[Function]}
  title="Question1"
>
  Question1
</a>
`;
